swagger.yaml

```diff
@@ -2273,6 +2273,70 @@ paths:
           schema:
             $ref: '#/definitions/error'
 
+  /clusters/{cluster_id}/logs_progress:
+    put:
+      tags:
+        - installer
+      security:
+        - agentAuth: []
+      description: Update log collection state and progress.
+      operationId: UpdateLogsProgress
+      parameters:
+        - in: path
+          name: cluster_id
+          description: The cluster whose log progress is being updated.
+          type: string
+          format: uuid
+          required: true
+        - in: query
+          name: logs_type
+          description: The type of log file.
+          type: string
+          enum: ['host', 'controller']
+          required: true
+        - in: query
+          name: host_id
+          description: The host whose log progress is being updated.
+          type: string
+          format: uuid
+          required: false
+        - in: query
+          name: logs_state
+          description: The state of collecting logs.
+          type: string
+          enum: ['requested', 'collecting', 'completed']
```

Comment:
Why do we need an API for `requested`? 
Shouldn't that get set by the service?

Reply:
Initially this was the intention, but since the remote components are not synchronized with the service there are times where the best place to indicate starting of log collection is in the remote component.

Reply:
Isn't `starting of log collection` == `collecting`?

Reply:
No. It's designed for controller logs (or the likes in the future) where the logs are continuously being uploaded. It parallel to FileUploaded and being updated from there